src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferTypedStringBuilderAppendOverloadsTests.cs

```diff
@@ -0,0 +1,273 @@
+                using System.Text;
+
+                class C
+                {
+                    public void M1() => new StringBuilder().Append(" + value + @".ToString());
+                    public void M2() => new StringBuilder().Insert(42, " + value + @".ToString());
+                }", diagnostics);
+        }
+
+        [Theory]
+        [InlineData("42")]
+        [InlineData("\"hello\"")]
+        [InlineData("DayOfWeek.Monday")]
+        [InlineData("DateTime.Now")]
+        public async Task NoDiagnostic_NoToStringCall(string value)
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+                using System;
+                using System.Text;
+
+                class C
+                {
+                    public void M1() => new StringBuilder().Append(" + value + @");
+                    public void M2() => new StringBuilder().Insert(42, " + value + @");
+                }");
+        }
+
+        [Theory]
+        [InlineData("42")]
+        [InlineData("DayOfWeek.Monday")]
+        [InlineData("DateTime.Now")]
+        public async Task NoDiagnostic_FormattedToString(string value)
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+                using System;
+                using System.Globalization;
+                using System.Text;
+
+                class C
+                {
+                    public void M1()
+                    {
+                        new StringBuilder()
+                            .Append(" + value + @".ToString(""X4""))
+                            .Append(" + value + @".ToString(CultureInfo.CurrentCulture))
+                            .Append(" + value + @".ToString(""X4"", CultureInfo.CurrentCulture))
+                            .Append(((IFormattable)" + value + @").ToString(""X4"", CultureInfo.CurrentCulture))
+                            .Append(((IFormattable)" + value + @").ToString());
+                    }
+
+                    public void M2()
+                    {
+                        new StringBuilder()
+                            .Insert(1, " + value + @".ToString(""X4""))
+                            .Insert(1, " + value + @".ToString(CultureInfo.CurrentCulture))
+                            .Insert(1, " + value + @".ToString(""X4"", CultureInfo.CurrentCulture))
+                            .Insert(1, ((IFormattable)" + value + @").ToString(""X4"", CultureInfo.CurrentCulture))
+                            .Insert(1, ((IFormattable)" + value + @").ToString());
+                    }
+                }");
+        }
+
+        [Fact]
+        public async Task NoDiagnostic_NotRelevantMethod()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+                using System;
+                using System.Text;
+
+                class C
+                {
+                    public void M()
+                    {
+                        var sb = new StringBuilder();
+                        sb.AppendLine(42.ToString());
```

Comment:
Is it correct/worth having this analyzer/fixer report/replace this pattern with `sb.Append(42).AppendLine()`?

Disregarding other pros/cons of the fluent builder pattern, it does happen to work here as a "one expression for another without side effects" solution.

Reply:
Interesting.  That seems reasonable.  I'd like to look at it separately.